python-checks/src/main/java/org/sonar/python/checks/hotspots/SecureCookieCheck.java

```diff
@@ -0,0 +1,167 @@
+public class SecureCookieCheck extends PythonSubscriptionCheck {
+
+  private static final String MESSAGE = "Make sure creating this cookie without the \"secure\" flag is safe.";
+  private static final Set<String> SET_COOKIE_METHODS = new HashSet<>(Arrays.asList(
+    "django.http.HttpResponse.set_cookie",
+    "django.http.HttpResponse.set_signed_cookie",
+    "django.http.HttpResponseRedirect.set_cookie",
+    "django.http.HttpResponseRedirect.set_signed_cookie",
+    "django.http.HttpResponsePermanentRedirect.set_cookie",
+    "django.http.HttpResponsePermanentRedirect.set_signed_cookie",
+    "django.http.HttpResponseNotModified.set_cookie",
+    "django.http.HttpResponseNotModified.set_signed_cookie",
+    "django.http.HttpResponseBadRequest.set_cookie",
+    "django.http.HttpResponseBadRequest.set_signed_cookie",
+    "django.http.HttpResponseNotFound.set_cookie",
+    "django.http.HttpResponseNotFound.set_signed_cookie",
+    "django.http.HttpResponseForbidden.set_cookie",
+    "django.http.HttpResponseForbidden.set_signed_cookie",
+    "django.http.HttpResponseNotAllowed.set_cookie",
+    "django.http.HttpResponseNotAllowed.set_signed_cookie",
+    "django.http.HttpResponseGone.set_cookie",
+    "django.http.HttpResponseGone.set_signed_cookie",
+    "django.http.HttpResponseServerError.set_cookie",
+    "django.http.HttpResponseServerError.set_signed_cookie",
+    "flask.Response.set_cookie"
+  ));
+
+  private static final String SECURE = "secure";
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Kind.ASSIGNMENT_STMT, ctx -> {
+      AssignmentStatement assignment = (AssignmentStatement) ctx.syntaxNode();
+      getSubscriptionToCookies(assignment.lhsExpressions())
+        .forEach(sub -> {
+          if (isSettingSecureFlag(sub) && isFalsy(assignment.assignedValue())) {
+            ctx.addIssue(assignment, MESSAGE);
+          }
+        });
+    });
+
+    context.registerSyntaxNodeConsumer(Kind.CALL_EXPR, ctx -> {
+      CallExpression callExpression = (CallExpression) ctx.syntaxNode();
+      Symbol calleeSymbol = callExpression.calleeSymbol();
+      if (calleeSymbol != null && SET_COOKIE_METHODS.contains(calleeSymbol.fullyQualifiedName())) {
+        if (callExpression.arguments().stream().anyMatch(argument -> argument.is(Kind.UNPACKING_EXPR))) {
+          return;
+        }
+        RegularArgument secureArgument = getSecureArgument(callExpression.arguments());
+        if (secureArgument == null || isFalsy(secureArgument.expression())) {
+          ctx.addIssue(callExpression.callee(), MESSAGE);
+        }
+      }
+    });
+  }
+
+  @CheckForNull
+  private static RegularArgument getSecureArgument(List<Argument> arguments) {
+    return arguments.stream()
+      .map(RegularArgument.class::cast)
+      .filter(argument -> {
+        Name keywordArgument = argument.keywordArgument();
+        return keywordArgument != null && keywordArgument.name().equals(SECURE);
```

Comment:
This implementation seems to assume the secure flag will always be passed as a keyword argument, with a risk of FP if it's passed as a positional argument. I haven't checked for django but I believe it's possible to use positional arguments with flask at least.
Would you consider accounting for positional arguments as well to avoid those FP (maybe by doing something similar to `PredictableSaltCheck`) ?

Reply:
Yes good point, I'll check for positional argument